### DISCUSSION TITLE
Discounts not working (with QBO)

### DISCUSSION THREAD

**Discussion by hhelmric** [codatio/codat-docs#951] [Q&A]
I'm trying to push an invoice from Codat to QBO with line-level discounts, but it seems QBO is overwriting the line-level `subTotals` with it's own calculation of `unitAmount * quantity` without looking at the provided `subTotal`, `totalAmount`, `discountPercentage`, or `discountAmount`

Essentially, if the Codat invoice has the following:
```
quantity: 1
unitAmount: 100
discountPercentage: 10
subTotal: 90
totalAmount: 90
```

when I view that invoice in QBO it will show a line totaling $100, not $90. I've tried many manipulations of subTotal, totalAmount, discountAmount, and discountPercentage and nothing will change the line total that shows in QBO.

Are we expected to edit the unit price with a post-discount-applied price? If so, then what's even the point of the `subTotal`, `totalAmount`, `discountAmount`, `discountPercentage` fields?

Additionally, the Codat documentation states that subTotals are _inclusive_ of discounts, but this release note has a conflicting explanation in the SubTotal rules section:
https://docs.codat.io/updates/220826-wbo-invoice-push

**Comment by dcoplowe** [comment]
Hi @hhelmric,

Sorry for the delay. I'd suggest reaching out to our support team (support@codat.io). They'll be able to get you the right answer and in a timely manner. 

Many thanks,

David
